client/app/components/dynamic-form/dynamicFormHelper.js

```diff
@@ -42,26 +43,32 @@ function normalizeSchema(configurationSchema) {
     }
 
     prop.required = includes(configurationSchema.required, name);
+    prop.advanced = includes(configurationSchema.advanced, name);
   });
 
   configurationSchema.order = configurationSchema.order || [];
 }
 
-function setDefaultValueForCheckboxes(configurationSchema, options = {}) {
-  if (Object.keys(options).length === 0) {
-    const properties = configurationSchema.properties;
-    Object.keys(properties).forEach((property) => {
-      if (!isUndefined(properties[property].default) && properties[property].type === 'checkbox') {
+function setDefaultValueToFields(configurationSchema, options = {}) {
+  const properties = configurationSchema.properties;
+  Object.keys(properties).forEach((property) => {
+    if (!isUndefined(properties[property].default)) {
+      // set default value for checkboxes and for advanced options
+      if (properties[property].type === 'checkbox' || properties[property].advanced) {
```

Comment:
Why the advanced options need special handling here?

Reply:
This is related to the second note 🙂:

- Default values for advanced options are filled in the form when you are creating a DS/Destination (regular options are turned into placeholders). This allows the user to fill in the form without even looking into the advanced options;

Reply:
We show as a placeholder the default value and we use it as well. So there is no need to have a different behavior for the advanced options: they can skip them and the default value will still be used.

Although this made me think of why aren't we just filling in the default value for regular options as well? The only reason I can think of is to be able to change the default, but not sure if this is desired :thinking: 

Anyway, it's out of scope, so worth keeping the same behavior until we decide if we want to change.

Reply:
I was thinking about required values also being able to be additional settings, but considering what you mentioned I shouldn't be worried about them at all (also not a very important context anyway)

Reply:
A required setting will never be in advanced settings. The idea is to hide there all the things people won't really bother configure in normal cases. This doesn't apply to required settings.